### ISSUE TITLE
texture mesh

### ISSUE THREAD

**Issue by LHXhh** [l1346792580123/diff#4]
作者您好，想请问下我该怎么做才能得到对应mesh的纹理图呢？

**Comment by l1346792580123** [comment]
纹理信息存在对应的_tex.pt文件里，你需要先读取纹理生成顶点带颜色的obj文件，再按照https://steamcommunity.com/sharedfiles/filedetails/?id=1265761945的做法将顶点颜色转成纹理图

**Comment by LHXhh** [comment]
很感谢您的回复，我想请问下得到的这个verts_tex是每个顶点的（1，num_points, 7）特征向量，我该怎么提取对应的顶点颜色呢，就是不太清楚verts_tex怎么用？真诚地希望能够得到您的帮助
`with torch.no_grad():
                vertices, faces, v, psr_grid, points = sap_generate(dpsr, psr2mesh, inputs, center, scale)
                verts_tex = grid_interp(tex_grid, v)

                sample_verts = v.squeeze(0).detach().cpu().numpy()
                save_verts = vertices.squeeze(0).detach().cpu().numpy()
                np_faces = faces.squeeze(0).detach().cpu().long().numpy()
                save_mesh = trimesh.Trimesh(save_verts, np_faces, process=False, maintain_order=True)
                save_mesh.export('ret/%d.obj'%idx)
                meshcleaning('ret/%d.obj'%idx)`

**Comment by l1346792580123** [comment]
前三维是漫反射颜色，第四维到第六维是镜面反射颜色，第七维是物体粗糙程度，你可以分别生成对应的贴图

**Comment by LHXhh** [comment]
好的，很感谢。但是当我将verts_tex的漫反射和镜面反射分别提取出来时，发现在训练过程的一半时还有颜色信息，虽然效果很糟糕，但是训练结束时颜色值归0了。模型几何还可以，但没有了顶点颜色。我是完全遵照源代码和所给的数据集进行训练的，我看论文上的texture mesh还可以，您知道是什么缘故呢？抱歉打扰您了
训练到一半
![image](https://github.com/l1346792580123/diff/assets/102360620/0acf8f02-8827-43aa-bff5-8d79d8da5a77)
训练结束时
![image](https://github.com/l1346792580123/diff/assets/102360620/e62ccf53-e525-4c21-9f8f-092df81a69c3)
